store/tikv/client.go

```diff
@@ -48,102 +58,266 @@ const (
 )
 
 type rpcClient struct {
-	msgID uint64
-	p     *Pools
+	p *Pools
 }
 
 func newRPCClient() *rpcClient {
 	return &rpcClient{
-		msgID: 0,
 		p: NewPools(maxConnection, func(addr string) (*Conn, error) {
 			return NewConnection(addr, dialTimeout)
 		}),
 	}
 }
 
-// SendCopReq sends a Request to co-processor and receives Response.
-func (c *rpcClient) SendCopReq(ctx goctx.Context, addr string, req *coprocessor.Request, timeout time.Duration) (*coprocessor.Response, error) {
-	start := time.Now()
-	defer func() { sendReqHistogram.WithLabelValues("cop").Observe(time.Since(start).Seconds()) }()
-
-	select {
-	case <-ctx.Done():
-		return nil, ctx.Err()
-	default:
-	}
+func (c *rpcClient) callRPCFunc(addr string, f func(conn *Conn), label string) error {
+	startTime := time.Now()
+	defer func() { sendReqHistogram.WithLabelValues(label).Observe(time.Since(startTime).Seconds()) }()
 
 	conn, err := c.p.GetConn(addr)
 	if err != nil {
-		return nil, errors.Trace(err)
+		return err
 	}
 	defer c.p.PutConn(conn)
-	msg := msgpb.Message{
-		MsgType: msgpb.MessageType_CopReq,
-		CopReq:  req,
+	f(conn)
+	return nil
+}
+
+func (c *rpcClient) KvGet(ctx goctx.Context, addr string, req *kvrpcpb.GetRequest) (*kvrpcpb.GetResponse, error) {
+	var resp *kvrpcpb.GetResponse
+	var err error
+	f := func(conn *Conn) {
+		client := tikvpb.NewTiKVClient(conn.ClientConn)
+		resp, err = client.KvGet(ctx, req)
+		if err != nil {
+			conn.Close()
+			err = errors.Trace(err)
+		}
```

Comment:
why not let the function return error directly? so we don't need to change the outer error explicitly.